planner/core/logical_plan_builder.go

```diff
@@ -1833,8 +1833,11 @@ func extractLimitCountOffset(ctx sessionctx.Context, limit *ast.Limit) (count ui
 	return count, offset, nil
 }
 
-func (b *PlanBuilder) buildLimit(src LogicalPlan, limit *ast.Limit) (LogicalPlan, error) {
-	b.optFlag = b.optFlag | flagPushDownTopN
+func (b *PlanBuilder) buildLimit(src LogicalPlan, limit *ast.Limit, calcFoundRows bool) (LogicalPlan, error) {
+	if !calcFoundRows {
+		// Only pushdown limit when calcFoundRows is not set.
+		b.optFlag = b.optFlag | flagPushDownTopN
+	}
```

Comment:
We should prevent the limit from being pushed down to the coprocessor as well? and we should not embed the limit into `IndexLookUp` either, please see `sinkIntoIndexLookUp()` for details.